### PR TITLE
[fix] #4253: Check genesis pub key in genesis round

### LINKED ISSUE #4253: Genesis is signed with correct key
When genesis is received via listen for genesis via `sumeragi_listen_for_genesis`(`BlockCreated`/`BlockSyncUpdate`) it is not verified that genesis is signed with `IROHA_GENESIS_PUBLIC_KEY`. Not even leader that creates the genesis block verifies that genesis is signed with correct key in `sumeragi_init_commit_genesis`

### PR DESCRIPTION
**Pull request by Erigara**
## Description

Verify that genesis transaction is indeed signed by genesis. 

### Linked issue

Closes #4253 

### Benefits

Not possible to submit wrong genesis.

### CODE REVIEW

**Review by s8sato — DISMISSED** [review]

File: core/src/block.rs [hunk 1]
```diff
@@ -275,6 +275,7 @@ mod valid {
             block: SignedBlock,
```
Comment (DCNick3) [inline core/src/block.rs:275]: ^ I suggest to add the new error condition into the doc above

`- Transaction in the genesis block is not signed by the genesis public key`

**Comment by DCNick3** [comment]
Other that this small nit, LGTM

**Comment by Erigara** [comment]
> do we have a test for this?

Don't think so, i've tested this manually as described [here](https://github.com/hyperledger/iroha/issues/4253#issuecomment-1929742514).

**Comment by mversic** [comment]
> > do we have a test for this?
> 
> Don't think so, i've tested this manually as described [here](https://github.com/hyperledger/iroha/issues/4253#issuecomment-1929742514).

My fear is that we could easily regress to this bug again

**Comment by Erigara** [comment]
I will add the test.

**Review by mversic — DISMISSED** [review]

**Review by DCNick3 — APPROVED** [review]

**Review by mversic — APPROVED** [review]

**Pull request merged**